-1

Tested at: e31907be3d3a0f4c37c0a881b9e46c507c566845

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16761/16615/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found an error when building: 
 
 <pre>>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2016-11-25-2300/src/Utilities/StaticAnalyzers/src/FiniteMathChecker.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2016-11-25-2300/src/Utilities/StaticAnalyzers/src/ConstCastChecker.cpp 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2016-11-25-2300/src/Utilities/StaticAnalyzers/src/ArgSizeChecker.cpp 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2016-11-25-2300/src/Utilities/StaticAnalyzers/src/ConstCastAwayChecker.cpp 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2016-11-25-2300/src/RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZ_vect.cc: In member function 'double DAClusterizerInZ_vect::update(double, DAClusterizerInZ_vect::track_t&, DAClusterizerInZ_vect::vertex_t&, bool, double&) const':
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2016-11-25-2300/src/RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZ_vect.cc:191:43: error: 'isNotFinite' was not declared in this scope
     if (isNotFinite(gtracks._Z_sum[itrack])) gtracks._Z_sum[itrack] = 0.0;
                                           ^
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2016-11-25-2300/src/RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZ_vect.cc:191:43: note: suggested alternative:
In file included from /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2016-11-25-2300/src/RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZ_vect.cc:10:0:
/cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_0_X_2016-11-25-2300/src/FWCore/Utilities/interface/isFinite.h:10:8: note:   'edm::isNotFinite'
</pre>

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 16 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2016-11-25-2300/src/RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZ.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2016-11-25-2300/src/RecoVertex/PrimaryVertexProducer/src/PrimaryVertexProducerAlgorithm.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2016-11-25-2300/src/RecoVertex/PrimaryVertexProducer/src/GapClusterizerInZ.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2016-11-25-2300/src/RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZ_vect.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2016-11-25-2300/src/RecoVertex/PrimaryVertexProducer/plugins/PrimaryVertexProducer.cc 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2016-11-25-2300/src/RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZ_vect.cc:191:9: error: use of undeclared identifier 'isNotFinite'; did you mean 'edm::isNotFinite'?
    if (isNotFinite(gtracks._Z_sum[itrack])) gtracks._Z_sum[itrack] = 0.0;
        ^~~~~~~~~~~
        edm::isNotFinite
/cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_0_X_2016-11-25-2300/src/FWCore/Utilities/interface/isFinite.h:10:8: note: 'edm::isNotFinite' declared here
  bool isNotFinite(T x) {
</pre>

--- a/RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZ_vect.cc
+++ b/RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZ_vect.cc
@@ -188,7 +188,7 @@ double DAClusterizerInZ_vect::update(double beta, track_t & gtracks,
     local_exp_list(gvertices._ei_cache, gvertices._ei, nv);
     
     gtracks._Z_sum[itrack] = kernel_add_Z(gvertices);
-    if (isNotFinite(gtracks._Z_sum[itrack])) gtracks._Z_sum[itrack] = 0.0;
+    if (edm::isNotFinite(gtracks._Z_sum[itrack])) gtracks._Z_sum[itrack] = 0.0;
     // used in the next major loop to follow
     if (!useRho0)
       sumpi += gtracks._pi[itrack];

